evalml/automl/automl_search.py

```diff
@@ -1006,6 +1009,24 @@ def search(self, show_iteration_plot=True):
                 f"Best pipeline {self.objective.name}: {best_pipeline['validation_score']:3f}"
             )
         self._searched = True
+        if self.search_iteration_plot is not None:
+            # go = import_or_raise(
+            #     "plotly.graph_objects",
+            #     error_msg="Cannot find dependency plotly.graph_objects",
+            # )
+            # title = "Pipeline Search: Iteration vs. {}<br><sub>Gray marker indicates the score at current iteration</sub>".format(
+            #     self.objective.name
+            # )
+            # layout = {
+            #     "title": title,
+            #     "xaxis": {"title": "Iteration", "rangemode": "tozero"},
+            #     "yaxis": {"title": "Score"},
+            # }
+            if not show_iteration_plot:
+                self.search_iteration_plot = self.plot.search_iteration_plot(
+                    interactive_plot=show_iteration_plot
+                )
```

Comment:
I had a really hard time understanding what this does. After falling down an iteration_plot rabbit hole, I think we should rename the `show_iteration_plot` argument to `interactive_plot`, to match the arg that's passed into `search_iteration_plot`. 

In my mind, that's how this logic makes sense. If `interactive_plot` is True, we call `self.plot.search_iteration_plot` up around line 910, which will create the interactive plot as we know and love. If `interactive_plot` is False and `verbose` is True (the other scenario where we'd want a plot), we run this code at the end to generate that final post-search plot that the docs require.

Could you refactor this section and the section around line 910 to follow that logic? It'd keep things much simpler and easier to follow.

Reply:
To be a little clearer here, I think this section could look like 
```
if self.verbose and not self.interactive_plot:
    self.search_iteration_plot = self.plot.search_iteration_plot(interactive_plot=interactive_plot)
```